src/yb/master/ts_descriptor.cc

```diff
@@ -164,9 +164,17 @@ std::string TSDescriptor::placement_id() const {
   return placement_id_;
 }
 
-void TSDescriptor::UpdateHeartbeatTime() {
+void TSDescriptor::UpdateHeartbeat(const TSHeartbeatRequestPB* req) {
   std::lock_guard<decltype(lock_)> l(lock_);
   last_heartbeat_ = MonoTime::Now();
+
+  DCHECK_GE(req->num_live_tablets(), 0);
```

Comment:
It seems these two checks can be lifted ahead of taking the lock

Reply:
Good catch, thanks! Fix committed.

Reply:
nit: you can enclose the lines starting with std::lock_guard to the last line in a pair of curly braces.

You can find many examples of such coding style throughout the YB code base.

I or one of my colleagues would put your patch through internal Jeninks.

Thanks for your contribution.

Reply:
Fixed. Thanks for your feedback! It's exciting to contribute to this great project 😆

Reply:
I submitted your patch for YB internal Jenkins.

Once test result comes back green and one of my co-workers confirms, this would be merged.